docs/developer-guides/native-token-migration-guide.md

```diff
@@ -0,0 +1,70 @@
+* `illiquid supply validator address` - this is the address to which the native tokens are sent on the 
+main chain to lock them so that they can be unlocked on the Partner Chain. This address is derived
+from the sidechain params and is present in the output of the `addresses` command of the
+`partner-chains-smart-contracts` CLI under `addresses/IlliquidCirculationSupplyValidator`.
+When using the `parner-chains-cli` wizard, it will be automatically set up in the `prepare-configuration` step.
+* native token minting `policy ID` and `asset name` - those should come from the native Cardano asset
+used to represent the Partner Chain's token on main chain. Because each native asset used can have
+different logic in its minting policy, development and creation of the asset is left for each
+Partner Chain developer team.
+When using the `parner-chains-cli` wizard, these can be set up in the `prepare-configuration` step.
+
+## Migration scenario - adding the feature
+
+1. Add the pallet into the runtime. This requires implementing the trait `TokenTransferHandler`, which
+is left for the developers of each particular Partner Chain to implement according to their needs and
+ledger structure. Consult the implementation in `runtime/src/lib.rs` for an example with a mocked handler.
+2. Bump the runtime spec version and note its new value, we will refer to it as `setup_version`.
+3. Wire the data source into the node.
+Consult the reference implementation in `node/src/main_chain_follower.rs` for an example.
+4. Add the inherent data provider to your inherent data provider creation logic.
+Consult the implementation in `node/src/inherent_data.rs` for an example.
+Because the inherent data provider should only run after the pallet is added and fully configured
+in the runtime, `new_for_runtime_version` factory function should be used, which accepts an additional predicate
+as argument, making it possible to specify the correct runtime version range:
+in this case `|version| version.spec_version > setup_version` should be used.
+5. Upgrade the nodes running the chain to the new node version containing the IDP added in previous steps.
+6. Perform a runtime upgrade (using `sudo` or other governance feature) to the spec version `setup_version`
+containing the pallet.
+7. Invoke the `set_main_chain_scripts` extrinsic on the newly added pallet, using the governance mechanism,
+to set the native token `policy ID` and `asset name`, and the `illiquid supply validator address`. After
+this step all information necessary for the IDP to operate is present.
+8. Pick a runtime spec version `launch_version` higher than `setup_version`.
+9. Perform another runtime upgrade, bumping the runtime spec version to `launch_version`.
+Once the upgrade is done, the native token data provider will start producing the inherent data based
+on observed native token transfers, triggering the pallet's inherent when necessary. 
```

Comment:
These start/stop versions add some complexity.
If `main_chain_scripts` were optional, and the IDP would do nothing if the config is not present, there would be only one runtime upgrade (to include the pallet) required. Point 4 becomes simpler. Points 5 and 6 order is arbitrary (but currently it seems the same). Points 8 and 9 are not present.

Reply:
yep, good idea. I added those changes in a separate Pr: https://github.com/input-output-hk/partner-chains/pull/109 . I'll update the description here